employees/forms.py

```diff
@@ -46,6 +50,8 @@ class Meta:
         fields = ("date", "description", "task_activities", "project", "work_hours")
         widgets = {"date": DatePickerInput(format="%Y-%m-%d")}
 
-    def __init__(self, *args: Any, **kwargs: Any):
-        super(ReportForm, self).__init__(*args, **kwargs)
-        self.fields["project"].queryset = kwargs["instance"].author.projects.all()
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        user = kwargs.pop("user", None)
```

Comment:
In this form you are passing `user` through `kwargs`, in previous `queryset` as extra parameter. Please unify this. I suggest that if element is required for the form it can be extra parameter, if it is optional it could be in `kwargs`. Also, with current code if you don't pass `user` variable will be set to None and line https://github.com/Code-Poets/sheetstorm/pull/274/files#diff-8e5e768a1bdc111c08d6403f77441ee3R56 will raise exception.

Btw. is this necessary to set both `queryset` and `choices` on choice field? I think it depends on if it is `ChoiceField` or `ModelChoiceField`. Please use only one if possible.

Reply:
Okay so I will add it as a additional parameter. But how we can set `user` variable to None? I am taking user from request or report author which is required during creating report.

I was wondering about it, and during adapt test I found that if I set queryset only, then my response doesn't have this information about what is in choicefield so I can't test it but when I runserver it is working fine and it displays all user projects. In other way when I set only choices then I have got this information in reponse, however there is possibility to add report by `POST` when you are not in some project. It looks like it works without  validation or something.

Reply:
> But how we can set user variable to None? I am taking user from request or report author which is required during creating report.

You are but you don't know in which circumstances someone else could use this form, and if he would know/remember to instantiate it correctly. You have to assume that other devs could use this form in wrong way, and don't test it ;) By making it required parameter you enforce them to use it correctly.

> during adapt test I found that if I set queryset only, then my response doesn't have this information about what is in choicefield so I can't test it but when I runserver it is working fine and it displays all user projects.

Which test and what was the problem? I have removed `self.fields["project"].choices = [(project.id, project.name) for project in user.projects.all()]` and all tests passed for me.